File: tac-1/programmable/programmable.ts
import { $ } from "bun";
import { readFileSync } from "fs";

async function main() {
    try {
        const promptContent = readFileSync("programmable/prompt.md", "utf-8");

        const output = await $`claude -p ${promptContent}`.text();

        console.log(output);
    } catch (error) {
        console.error(`Error executing command: ${error}`);
        process.exit(1);
    }
}

if (import.meta.main) {
    main();
}

File: tac-1/programmable/programmable.py
import subprocess
import sys


def main():
    with open("programmable/prompt.md", "r") as f:
        prompt_content = f.read()

    command = [
        "claude",
        "-p",
        prompt_content,
    ]

    try:
        result = subprocess.run(command, capture_output=True, text=True)

        print(result.stdout)

        if result.stderr:
            print(result.stderr, file=sys.stderr)

        sys.exit(result.returncode)

    except Exception as e:
        print(f"Error executing command: {e}", file=sys.stderr)
        sys.exit(1)


if __name__ == "__main__":
    main()
